plugins/destination_testing.go

```diff
@@ -155,53 +206,61 @@ func (s *destinationTestSuite) destinationPluginTestWriteAppend(ctx context.Cont
+		return err
 	}
 
-	resource = schema.DestinationResource{
-		TableName: table.Name,
-		Data:      testdata.TestData(),
+	if err := p.writeOne(ctx, tables, sourceName, syncTime, resource1); err != nil {
+		return fmt.Errorf("failed to write one second time: %w", err)
 	}
 
-	if !s.tests.SkipSecondAppend {
-		// we dont use time.now because looks like there is some strange
-		// issue on windows machine on github actions where it returns the same thing
-		// for all calls.
-		secondSyncTime := syncTime.Add(time.Second).UTC()
-		// write second time
-		if err := p.writeOne(ctx, tables, sourceName, secondSyncTime, resource); err != nil {
-			return fmt.Errorf("failed to write one second time: %w", err)
-		}
+	// we dont use time.now because looks like there is some strange
+	// issue on windows machine on github actions where it returns the same thing
+	// for all calls.
+	secondSyncTime := syncTime.Add(time.Second).UTC()
+	// write second time
+	if err := p.writeOne(ctx, tables, sourceName, secondSyncTime, resource2); err != nil {
+		return fmt.Errorf("failed to write one second time: %w", err)
 	}
 
 	resourcesRead, err := p.readAll(ctx, tables[0], sourceName)
 	if err != nil {
 		return fmt.Errorf("failed to read all second time: %w", err)
 	}
 
-	expectedResource := 2
-	if s.tests.SkipSecondAppend {
-		expectedResource = 1
+	if len(resourcesRead) != 2 {
+		return fmt.Errorf("expected 2 resources, got %d", len(resourcesRead))
 	}
 
-	if len(resourcesRead) != expectedResource {
-		return fmt.Errorf("expected %d resources, got %d", expectedResource, len(resourcesRead))
+	expectedResourceData := []schema.CQTypes{
+		appendSourceNameAndSyncTimeToCqTypes(sourceName, syncTime, resource1.Data),
+		appendSourceNameAndSyncTimeToCqTypes(sourceName, secondSyncTime, resource2.Data),
 	}
 
-	if resource.Data.Equal(resourcesRead[0]) {
-		return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[0])
+	sort.Slice(expectedResourceData, func(i, j int) bool {
+		// We add '+2' to uuidFieldIndex because we add 2 columns (_cq_sync_time and source_name)
+		return expectedResourceData[i][cqSyncTimeFieldIndex+2].String() < expectedResourceData[j][cqSyncTimeFieldIndex+2].String()
+	})
+	sort.Slice(resourcesRead, func(i, j int) bool {
+		// We add '+2' to uuidFieldIndex because we add 2 columns (_cq_sync_time and source_name)
+		return resourcesRead[i][cqSyncTimeFieldIndex+2].String() < resourcesRead[j][cqSyncTimeFieldIndex+2].String()
+	})
+
+	if !expectedResourceData[0].Equal(resourcesRead[0]) {
+		return fmt.Errorf("expected data to be %v, got %v", expectedResourceData[0], resourcesRead[0])
 	}
-	if !s.tests.SkipSecondAppend {
-		if resource.Data.Equal(resourcesRead[1]) {
-			return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[1])
-		}
+	if !expectedResourceData[1].Equal(resourcesRead[1]) {
+		return fmt.Errorf("expected data to be %v, got %v", expectedResourceData[1], resourcesRead[1])
```

Comment:
```suggestion
		return fmt.Errorf("expected data[1] to be %v, got %v", expectedResourceData[1], resourcesRead[1])
```

Readers of the test error should be able to differentiate it from the case above somehow